client.go

```diff
@@ -0,0 +1,788 @@
+func (c *Client) Incr(key string) *goja.Promise {
+	promise, resolve, reject := c.makeHandledPromise()
+
+	go func() {
+		newValue, err := c.redisClient.Incr(c.vu.Context(), key).Result()
+		if err != nil {
+			reject(err)
+		}
+
+		resolve(newValue)
+	}()
+
+	return promise
+}
+
+// IncrBy increments the number stored at `key` by `increment`. If the key does
+// not exist, it is set to zero before performing the operation. An
+// error is returned if the key contains a value of the wrong type, or
+// contains a string that cannot be represented as an integer.
+func (c *Client) IncrBy(key string, increment int64) *goja.Promise {
+	promise, resolve, reject := c.makeHandledPromise()
+
+	go func() {
+		newValue, err := c.redisClient.IncrBy(c.vu.Context(), key, increment).Result()
+		if err != nil {
+			reject(err)
+		}
+
+		resolve(newValue)
+	}()
+
+	return promise
+}
+
+// Decr decrements the number stored at `key` by one. If the key does
+// not exist, it is set to zero before performing the operation. An
+// error is returned if the key contains a value of the wrong type, or
+// contains a string that cannot be represented as an integer.
+func (c *Client) Decr(key string) *goja.Promise {
+	promise, resolve, reject := c.makeHandledPromise()
+
+	go func() {
+		newValue, err := c.redisClient.Decr(c.vu.Context(), key).Result()
+		if err != nil {
+			reject(err)
+		}
+
+		resolve(newValue)
+	}()
+
+	return promise
+}
+
+// DecrBy decrements the number stored at `key` by `decrement`. If the key does
+// not exist, it is set to zero before performing the operation. An
+// error is returned if the key contains a value of the wrong type, or
+// contains a string that cannot be represented as an integer.
+func (c *Client) DecrBy(key string, decrement int64) *goja.Promise {
+	promise, resolve, reject := c.makeHandledPromise()
+
+	go func() {
+		newValue, err := c.redisClient.DecrBy(c.vu.Context(), key, decrement).Result()
+		if err != nil {
+			reject(err)
+		}
+
+		resolve(newValue)
+	}()
+
+	return promise
+}
+
+// RandomKey returns a random key.
+func (c *Client) RandomKey() *goja.Promise {
+	promise, resolve, reject := c.makeHandledPromise()
+
+	go func() {
+		key, err := c.redisClient.RandomKey(c.vu.Context()).Result()
+		if err != nil {
+			reject(err)
+		}
+
+		resolve(key)
+	}()
+
+	return promise
+}
+
+// Mget returns the values associated with the specified keys.
+func (c *Client) Mget(keys ...string) *goja.Promise {
+	promise, resolve, reject := c.makeHandledPromise()
+
+	go func() {
+		values, err := c.redisClient.MGet(c.vu.Context(), keys...).Result()
+		if err != nil {
+			reject(err)
+		}
+
+		resolve(values)
+	}()
+
+	return promise
+}
+
+// Expire sets a timeout on key, after which the key will automatically
+// be deleted.
+// Note that calling Expire with a non-positive timeout will result in
+// the key being deleted rather than expired.
+func (c *Client) Expire(key string, seconds int) *goja.Promise {
```

Comment:
I guess given that Redis defines it that it takes seconds it will be strange for us to take something else.

It seems like there is [PEXPIRE](https://redis.io/commands/pexpire/) which takes milliseconds ... so I guess when/if we add it we will have it's argument as milliseconds :shrug: